### ISSUE TITLE
Failed to connect to the remote extension host server

### ISSUE THREAD

**Issue by Sysa** [microsoft/vscode-tools-for-ai#1419]
Issue Type: <b>Bug</b>

Hello VS Code support!

I tried to connect to remote AML computes instance (was fine last week), but can't do it right now, also restart of the compute instance- had no luck. I believe something is weird with either compute instance or an AML extension, but can't figure out what exactly, since there is not much info on it and VS code just can't start due to the exception.
Last week also moved from ADAP to MSAL authentication for an Azure ML project (`az cli` from 2.29 -> 2.32; `azureml-sdk` from 1.34 -> 1.37) - that also might be a root cause. The token date looks quite old. Can't do anything, only start to use a new compute instance.

**_Edit original post:_**
P.S. New compute instance didn't help, something wrong with vscode/extensions. **Issue solved with:** Disable all extensions via CLI run, Re-install VS code, File -> Close remote connection (after reinstallation over previous version), then enable back all extension and connect to the compute instance finally works. Root cause analysis is still unclear and also if there was a way to clean-up expired token or something.

Exception message:

Failed to connect to the remote extension host server (Error: invalid_grant: 700082 - [2022-01-24 15:19:04Z]: AADSTS700082: The refresh token has expired due to inactivity. The token was issued on 2021-10-26T14:09:08.3243385Z and was inactive for 90.00:00:00.
Trace ID: e669aa0f-c480-47e4-9961-28f827032b00
Correlation ID: 8ee46643-4907-4232-869b-a7d9048825fc
Timestamp: 2022-01-24 15:19:04Z - Correlation ID: 8ee46643-4907-4232-869b-a7d9048825fc - Trace ID: e669aa0f-c480-47e4-9961-28f827032b00)

VS Code version: Code 1.63.2 (899d46d82c4c95423fb7e10e68eba52050e30ba3, 2021-12-15T09:40:02.816Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No
Fetching remote diagnostics for 'Azure ML' failed: invalid_grant: 700082 - [2022-01-24 15:19:04Z]: AADSTS700082: The refresh token has expired due to inactivity. The token was issued on 2021-10-26T14:09:08.3243385Z and was inactive for 90.00:00:00.
Trace ID: e669aa0f-c480-47e4-9961-28f827032b00
Correlation ID: 8ee46643-4907-4232-869b-a7d9048825fc
Timestamp: 2022-01-24 15:19:04Z - Correlation ID: 8ee46643-4907-4232-869b-a7d9048825fc - Trace ID: e669aa0f-c480-47e4-9961-28f827032b00

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz (4 x 2594)|
|GPU Status|2d_canvas: unavailable_software<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: disabled_off<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software|
|Load (avg)|undefined|
|Memory (System)|16.00GB (5.98GB free)|
|Process Argv|--crash-reporter-id b268be4a-29ce-4024-994d-4b9cee2e47b0|
|Screen Reader|no|
|VM|0%|

Fetching remote diagnostics for 'Azure ML' failed: invalid_grant: 700082 - [2022-01-24 15:19:04Z]: AADSTS700082: The refresh token has expired due to inactivity. The token was issued on 2021-10-26T14:09:08.3243385Z and was inactive for 90.00:00:00.
Trace ID: e669aa0f-c480-47e4-9961-28f827032b00
Correlation ID: 8ee46643-4907-4232-869b-a7d9048825fc
Timestamp: 2022-01-24 15:19:04Z - Correlation ID: 8ee46643-4907-4232-869b-a7d9048825fc - Trace ID: e669aa0f-c480-47e4-9961-28f827032b00
</details>

<details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
pythonvspyl392:30425749
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dsc:30424895
pythonvs932:30410667
vscop804:30404766
vscop341:30404997
vsrem710cf:30416617
vsbas813:30426126
vsic-wlc-tsjs:30426436

```

</details>

<details>
<summary>λ code --list-extensions --show-versions</summary>

```

donjayamanne.githistory@0.6.19
ms-azuretools.vscode-docker@1.19.0
ms-python.python@2021.12.1559732655
ms-python.vscode-pylance@2022.1.3
ms-toolsai.jupyter@2021.11.1001550889
ms-toolsai.jupyter-keymap@1.0.0
ms-toolsai.jupyter-renderers@1.0.4
ms-toolsai.vscode-ai@0.6.27
ms-toolsai.vscode-ai-remote@0.6.0
ms-vscode-remote.remote-containers@0.209.6
ms-vscode-remote.remote-wsl@0.63.13
ms-vscode.azure-account@0.9.11
waderyan.gitblame@8.1.0
```

</details>

Similar issue found with no clear solution (but new instance creation) - https://github.com/microsoft/vscode-tools-for-ai/issues/860

**Comment by sevillal** [comment]
Hello @Sysa, thank you for filing this, we are so sorry to hear that you are facing this issue.

Some questions: 
1. Are you connection VS Code compute instance from the Azure ML Studio?
2. Can you try the following when you reach the failure:
    - Open VS Code: File -> Closer remote connection
    - Retry connecting to the Compute Instance

Are open to having a quick meeting to triage the issue? let me know and I can setup some time. :)

**Comment by Sysa** [comment]
Hi @sevillal !
Thank you for your prompt reply.

1. Yes, that's true.
2. Yes, that helped (see my updated initial message). But I found `Close remote connection` button only after re-installing of the VS Code. But the closing of the connection helped eventually.

I am open to having a call/meeting about triaging the issue (but I failed to reproduce so far), also let me know if an internal MSFT Support ticket is required to be filled for that. Feel free to archive as well, since it is resolved and have the solution (but with no root cause analysis).

**Comment by sevillal** [comment]
@Sysa thanks for trying that out.

Do you still need to do the steps you mentioned in you issue description to connect every time? (i.e. Disable all extensions via CLI run, Re-install VS code, File -> Close remote connection...)

If that's the case I think it would help triage further to help you solve that. What time zone are you at?

**Comment by Sysa** [comment]
Sorry for the delay @sevillal 
No, I did it once and after that actions it works like a charm without any issues.
I feel like there is no root cause analysis from my side or from logs, so probably that issue can be closed without a root cause findings.
I'm in CET.

**Comment by QiwenZheng1234** [comment]
> Hello @Sysa, thank you for filing this, we are so sorry to hear that you are facing this issue.
> 
> Some questions:
> 
> 1. Are you connection VS Code compute instance from the Azure ML Studio?
> 2. Can you try the following when you reach the failure:
>    
>    * Open VS Code: File -> Closer remote connection
>    * Retry connecting to the Compute Instance
> 
> Are open to having a quick meeting to triage the issue? let me know and I can setup some time. :)

Hi, I'm having the same problem. I closed remote connection, then retry connecting to the compute instance. A window popped up after I entering my password three times: 
![image](https://user-images.githubusercontent.com/60889372/200628951-b8edc047-8ed5-42f0-b20a-f8c72ab6f5f2.png)

**Comment by sevillal** [comment]
@QiwenZheng1234 thanks for you comment. I have created a[ new issue ](https://github.com/microsoft/vscode-tools-for-ai/issues/1791)to track it since it looks different from the original post. I have posted some questions for you in the new issue. 

https://github.com/microsoft/vscode-tools-for-ai/issues/1791
